WrightTools/artists/_helpers.py

```diff
@@ -0,0 +1,917 @@
+        Factor. Default is 200.
+
+    Returns
+    -------
+    ([h_scaled, v_scaled], [va, ha])
+    """
+    # get bounds
+    x0, y0, width, height = ax.bbox.bounds
+    width_scaled = width / factor
+    height_scaled = height / factor
+    # get scaled postions
+    if position == 'UL':
+        v_scaled = distance / width_scaled
+        h_scaled = 1 - (distance / height_scaled)
+        va = 'top'
+        ha = 'left'
+    elif position == 'LL':
+        v_scaled = distance / width_scaled
+        h_scaled = distance / height_scaled
+        va = 'bottom'
+        ha = 'left'
+    elif position == 'UR':
+        v_scaled = 1 - (distance / width_scaled)
+        h_scaled = 1 - (distance / height_scaled)
+        va = 'top'
+        ha = 'right'
+    elif position == 'LR':
+        v_scaled = 1 - (distance / width_scaled)
+        h_scaled = distance / height_scaled
+        va = 'bottom'
+        ha = 'right'
+    else:
+        print('corner not recognized')
+        v_scaled = h_scaled = 1.
+        va = 'center'
+        ha = 'center'
+    return [h_scaled, v_scaled], [va, ha]
+
+
+def pcolor_helper(xi, yi, zi):
+    """Prepare a set of arrays for plotting using `pcolor`.
+
+    The return values are suitable for feeding directly into ``matplotlib.pcolor``
+    such that the pixels are properly centered.
+
+    Parameters
+    ----------
+    xi : 1D or 2D array-like
+        Array of X-coordinates.
+    yi : 1D or 2D array-like
+        Array of Y-coordinates.
+    zi : 2D array-like
+        Rectangular array of Z-coordinates.
+
+    Returns
+    -------
+    X : 2D ndarray
+        X dimension for pcolor
+    Y : 2D ndarray
+        Y dimension for pcolor
+    Z : 2D ndarray
+        Z dimension for pcolor
+    """
+    xi = xi.copy()
+    yi = yi.copy()
+    zi = zi.copy()
+    if xi.ndim == 1:
+        xi.shape = (xi.size, 1)
+    if yi.ndim == 1:
+        yi.shape = (1, yi.size)
+    shape = wt_kit.joint_shape([xi, yi])
+    # full
+
+    def full(arr):
+        for i in range(arr.ndim):
+            if arr.shape[i] == 1:
+                arr = np.repeat(arr, shape[i], axis=i)
+        return arr
+
+    xi = full(xi)
+    yi = full(yi)
+    # pad
+    x = np.arange(shape[1])
+    y = np.arange(shape[0])
+    f_xi = interp2d(x, y, xi)
+    f_yi = interp2d(x, y, yi)
+    x_new = np.arange(-1, shape[1] + 1)
+    y_new = np.arange(-1, shape[0] + 1)
+    xi = f_xi(x_new, y_new)
+    yi = f_yi(x_new, y_new)
+    # fill
+    X = np.empty([s - 1 for s in xi.shape])
+    Y = np.empty([s - 1 for s in yi.shape])
+    for orig, out in [[xi, X], [yi, Y]]:
+        for idx in np.ndindex(out.shape):
+            ul = orig[idx[0] + 1, idx[1] + 0]
+            ur = orig[idx[0] + 1, idx[1] + 1]
+            ll = orig[idx[0] + 0, idx[1] + 0]
+            lr = orig[idx[0] + 0, idx[1] + 1]
+            out[idx] = np.mean([ul, ur, ll, lr])
+    return X, Y, zi
+
+
+def plot_colorbar(cax=None, cmap='default', ticks=None, clim=None, vlim=None,
```

Comment:
should this be in `_helpers` or `_colors`?

Reply:
I think I do like it in _helpers